ownCloud/Settings/MediaUploadSettingsViewController.swift

```diff
@@ -28,11 +29,41 @@ class MediaUploadSettingsViewController: StaticTableViewController {
 
 		if let userDefaults = OCAppIdentity.shared.userDefaults {
 			self.addSection(MediaExportSettingsSection(userDefaults: userDefaults))
+
+			let enterpriseAccountAvailable = OCBookmarkManager.shared.bookmarks.filter({$0.edition == .Enterprise}).count > 0
+
+			if enterpriseAccountAvailable || isProPhotoPackageLicensed() {
+				self.addSection(ProPhotoUploadSettingsSection(userDefaults: userDefaults))
+			}
+
 			if OCBookmarkManager.shared.bookmarks.count > 0 {
 				self.addSection(AutoUploadSettingsSection(userDefaults: userDefaults))
 				// TODO: Re-add this section when we re-gain an ability to run background NSURLSessions
 				//self.addSection(BackgroundUploadsSettingsSection(userDefaults: userDefaults))
 			}
 		}
 	}
+
+	private func isProPhotoPackageLicensed() -> Bool {
```

Comment:
The code in this method is using a snapshot of the current authorization status for that feature - and is dealing with an edge case as well. I see a few issues here:
- the UI only reflects the authorization status at the time the UI is invoked (consider iPad multi-window, where settings can be open while accounts get added or removed)
- handling the edge case here is externalizing part of the internal implementation
- the DispatchGroup could put the app at risk of becoming unresponsive if `OCLicenseManager.shared.perform(afterCurrentlyPendingRefreshes:)` doesn't return promptly - or ever decides to dispatch the provided block to the main thread

Therefore, please use 
```swift
OCLicenseManager.shared.observeProducts(nil, features: [ .photoProFeatures ], in: OCLicenseEnvironment(), withOwner: self) { (licenseObserver, isInitial, authStatus) in
    // call code that adds and removes the section depending on authStatus and number of enterprise accounts
    reconsiderProPhotoSettingsSection()
}
```
instead, which abstracts away the edge case and won't block the main thread.

To also react to changes in the number of enterprise accounts, you could listen for `OCBookmarkManagerListChanged` notifications and call `reconsiderProPhotoSettingsSection()` (from the above example) on that occasion, too.